File: competition-serv-model/src/main/java/compman/compsrv/model/commands/payload/UpdateCategoryFightsPayload.java
package compman.compsrv.model.commands.payload;

import compman.compsrv.model.dto.schedule.MatScheduleContainerDTO;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@NoArgsConstructor
public class UpdateCategoryFightsPayload implements Serializable {
    private MatScheduleContainerDTO[] fightsByMats;
}

File: competition-serv-service/src/main/java/compman/compsrv/model/es/commands/payload/UpdateCategoryFightsPayload.kt
package compman.compsrv.model.es.commands.payload

import compman.compsrv.model.schedule.MatScheduleContainer

data class UpdateCategoryFightsPayload(val fightsByMats: Array<MatScheduleContainer>) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as UpdateCategoryFightsPayload

        if (!fightsByMats.contentEquals(other.fightsByMats)) return false

        return true
    }

    override fun hashCode(): Int {
        return fightsByMats.contentHashCode()
    }
}